benchmarks/views/index.py

```diff
@@ -30,13 +32,76 @@
+    'submission_id', 'jenkins_id', 'timestamp'
+])
+
+ScoreDisplay = namedtuple('ScoreDisplay', [
+    'benchmark', 'versioned_benchmark_identifier',
+    'score_raw', 'score_ceiled', 'error', 'color', 
+    'comment', 'is_complete'
+])
+
+def cache_get_context(timeout=24 * 60 * 60):  # 24 hour cache by default
+    '''
+    Decorator that caches results of get_context function for faster loading of leaderboard view and model card view.
+    Two-level caching:
+        - Global cache for public data
+        - User-specific cache for non-public data
+    Args:
+        timeout (int): Cache timeout in seconds. Defaults to 24 hours.
+    '''
+    def decorator(func):  # Take function to be decorated (i.e., get_context)
+        @wraps(func)  # Preserving original function's metadata attributes
+        def wrapper(user=None, domain="vision", benchmark_filter=None, model_filter=None, show_public=False):
+            # Determine which type of cache key to use
+            if show_public and not user:
+                # (CASE 1: Public data) Create unique key for public data cache
+                key_parts = ['global_context', domain, 'public']
+            elif user:
+                # (CASE 2: User data) Create unique key for user-specific cache
+                key_parts = ['user_context', domain, str(user.id), str(show_public)]
+            else:
+                # (CASE 3: No caching) Neither public nor user-specific
+                return func(user=user, domain=domain, benchmark_filter=benchmark_filter, 
+                          model_filter=model_filter, show_public=show_public)
+            
+            # Generate MD5 hash (to avoid invalid characters, too many characters, etc.). Not necessary but good practice.
+            cache_key = hashlib.md5('_'.join(key_parts).encode()).hexdigest()
+            
+            # Try to get cached result
+            cached_result = cache.get(cache_key)
+            if cached_result is not None:
+                return cached_result  # Return cached data if found
+
+            # If no cache found, calculate result as normal
+            result = func(user=user, domain=domain, benchmark_filter=benchmark_filter, 
+                        model_filter=model_filter, show_public=show_public)
+            
+            # Store result in cache appropriately (i.e., for users or globally)
+            cache.set(cache_key, result, timeout)
+            return result
+
+        return wrapper
+    return decorator
+
+# Keep leaderboard caching for now.
 @cache_page(24 * 60 * 60)
 def view(request, domain: str):
-    context = get_context(domain=domain)
-    return render(request, 'benchmarks/leaderboard/leaderboard.html', context)
-
-
+    # Pre-cache public context for model cards
+    public_context = get_context(domain=domain, show_public=True)
+    # Cache leaderboard context that is ultimately rendered in the leaderboard view
+    private_context = get_context(domain=domain)
```

Comment:
if this is the public (main) leaderboard, wouldn't we want `public=True` here as well? I get what you're trying to do, but I am a tad confused - maybe the name of the variables is throwing me off based on the comments above them

Reply:
I kept the main leaderboard `get_context` the same (https://github.com/brain-score/brain-score.web/blob/master/benchmarks/views/index.py#L36)

I think the confusion is what `show_public` is referring to which I think might be referring to something related to the competition models?

I'll change the name of `private_context` (which is definitely not appropriate) to `leaderboard_context`.